Port 8480 automation guide automate incident management with slack GitHub issues and port (#1307)
port-labs/port-docs: docs/guides-and-tutorials/create-slack-channel-for-reported-incident.md

- Send a breif message regarding the incident in the Slack channel for visibility.
- Create a Github issue for documenting the incident.

### Automation backend
As a backend for this automation, we will create a Github Workflow in our repository.

<details>
    <summary>`Handle incident` GitHub workflow YAML</summary>

This workflow is responsible for managing a new incident. It will be triggered via Port automation.

```yaml showLineNumbers title=".github/workflows/handle-incident.yaml"
name: Handle Incident

on:
  workflow_dispatch:
    inputs:
      port_payload:
        description: "Port's payload, including details for who triggered the action and general context (blueprint, run ID, etc...)."
        required: true

# These permissions are required for the Github issue creation
permissions:
  contents: read
  issues: write 

jobs:
  handle-new-incident:
    runs-on: ubuntu-latest
    env:
      PD_INCIDENT_ID: ${{ fromJson(inputs.port_payload).event.diff.after.identifier }}
      PD_INCIDENT_URL: ${{ fromJson(inputs.port_payload).event.diff.after.properties.url }}
      PD_INCIDENT_TITLE: ${{ fromJson(inputs.port_payload).event.diff.after.title }}

>       PORT_INCIDENT_URL: https://app.getport.io/pagerdutyIncidentEntity?identifier=${{ fromJson(inputs.port_payload).event.diff.after.identifier }}

Reviewer: Add highlight and comment here mentioning to select the correct Port region

    steps:
      - uses: actions/checkout@v4

      - name: Log Github Issue Creation
        uses: port-labs/port-github-action@v1
        with:
          clientId: ${{ secrets.PORT_CLIENT_ID }}
          clientSecret: ${{ secrets.PORT_CLIENT_SECRET }}
          baseUrl: https://api.getport.io
          operation: PATCH_RUN
          runId: ${{ fromJson(github.event.inputs.port_payload).run.id }}
          logMessage: "Creating a new Github issue for Pagerduty incident '${{ env.PD_INCIDENT_ID }}'..."

      - name: Get incident's related service
        id: get-incidient-service
        uses: port-labs/port-github-action@v1
        with:
          clientId: ${{ secrets.PORT_CLIENT_ID }}
          clientSecret: ${{ secrets.PORT_CLIENT_SECRET }}
          baseUrl: https://api.getport.io
          operation: GET
          blueprint: pagerdutyService
          identifier: ${{ fromJson(inputs.port_payload).event.diff.after.relations.pagerdutyService }}
      
      # The Github Service entity identifier is defined as Pagerduty title lowercased and split by '-'

>       - name: Extract realted service

Reviewer: ```suggestion
      - name: Extract related service
```

          SVC_ENTITY_TITLE: ${{ steps.get-service-info.outputs.SERVICE_TITLE }}
        id: send-message
        with:
          slack-function: send-message
          slack-bot-user-oauth-access-token: ${{ secrets.BOT_USER_OAUTH_TOKEN }}
          slack-channel: ${{ steps.create-slack-channel.outputs.SLACK_CHANNEL_ID }}
          slack-text: | 
            :rotating_light: New Incident reported - ${{ env.PD_INCIDENT_TITLE }} :rotating_light:
              Urgency: `${{ fromJson(inputs.port_payload).event.diff.after.properties.urgency }}`
              Service: <${{ env.SVC_ENTITY_URL }}|${{ env.SVC_ENTITY_TITLE }}>
              Manage incident :point_right::skin-tone-4: <${{ env.PORT_INCIDENT_URL }}|here>!

              Please use this Slack channel to report any updates, ideas, or root-cause ideas related to this incident :thread:

      - name: Update incident entity with new information
        uses: port-labs/port-github-action@v1
        env:
          SLACK_CHANNEL_URL: https://slack.com/app_redirect?channel=${{ steps.create-slack-channel.outputs.SLACK_CHANNEL_ID }}
        with:
          clientId: ${{ secrets.PORT_CLIENT_ID }}

>           clientSecret: ${{ secrets.PORT_CLIENT_SECRET }}

Reviewer: Missing `baseUrl` here

          identifier: ${{ env.PD_INCIDENT_ID }}
          blueprint: pagerdutyIncident
          properties: |
            {
              "slack_channel": "${{ env.SLACK_CHANNEL_URL }}"
            }
          relations: | 
            {
              "githubIssue": "${{ steps.get-service-info.outputs.SERVICE_IDENTIFIER }}-${{ steps.create-github-issue.outputs.number }}",
              "service": "${{ steps.get-service-info.outputs.SERVICE_IDENTIFIER }}"
            }

      - name: Log Successful Action
        if: success()
        uses: port-labs/port-github-action@v1
        with:
          clientId: ${{ secrets.PORT_CLIENT_ID }}
          clientSecret: ${{ secrets.PORT_CLIENT_SECRET }}
          baseUrl: https://api.getport.io
          operation: PATCH_RUN
          runId: ${{ fromJson(github.event.inputs.port_payload).run.id }}
          logMessage: |
            Done handling the new incident 💪🏻

> ```

Reviewer: Add here the template block about the Port region parameters

<p align="center">
<img src='/img/guides/slackIncidentGuide/newAutomationRun.png' width='50%' border='1px' />
</p>

5. Click on the run and view the logs. Wait for the run to be in `Success` state.

<!-- Run page - successful run -->
<p align="center">
<img src='/img/guides/slackIncidentGuide/successfulAutomationRun.png' width='75%' border='1px' />
</p>

6. Navigate back to your [Pagerduty Incidents](https://app.getport.io/pagerdutyIncidents) page. Click on the incident entity you ingested.
7. You should notice that the `Slack Channel URL` property, and the `Github Issue` relation are populated.

<!-- Incident entity page - Updated properties -->
<p align="center">
<img src='/img/guides/slackIncidentGuide/updatedIncidentEntity.png' width='75%' border='1px' />
</p>

8. Click the `Slack Channel URL`. This will allow you to join the dedicated Slack channel created for this incident.
9. Navigate to the `Github Issue` through the relation. Then navigate to the `Link` to view the Github issue created as part of the automation.

## Summary 
Using Port as the automation orchestrator, we created an incident management flow which helps us keep a high standard when facing new incidents.

> This automation will allow a faster notification time regarding new incidents, and a dedicated place to keep track of the troubleshooting process and any interesting updates.

Reviewer: ```suggestion
This automation will enable faster notification and response time when handling new incidents, and a dedicated place to keep track of the troubleshooting process and any relevant updates.
```

## Next Steps

> This guide can be enhanced to further meet your organization's needs. Here are some nice ideas you can implement:

Reviewer: ```suggestion
This guide can be enhanced to further meet your organization's needs. Here are some ideas you can implement:
```
